app/controllers/services_controller.rb

```diff
@@ -0,0 +1,64 @@
+class ServicesController < ApplicationController
+
+  # I think this means respond only to accept: application/json
+  # We might want HTML responses as well, and/or csv
+  respond_to :json
+
+  def authenticate_service
+    # User needs to be logged in already!
+    # Future: provide an automated service so scripts can fetch a token.
+    # n.b. token only gives access to services, not to the rest of the
+    # web site.
+    if current_user
+      if current_user.is_power_user?
+        render json: { token: jwt_token(current_user) }
+      else
+        return render_unauthorized errors: { forbidden: ["You are not authorized to use the web services."] }
+      end
+    else
+      return render_unauthorized errors: { unauthorized: ["Please log in, then try again."] }
+    end
+  end
+
+  protected
+
+  # No args - act as a guard on any web service action that wants authentication
+
+  def authorize_user_from_token!
+    if current_user
+      raise "Forbidden" unless @current_user.is_power_user?    # Unauthorized
+      @current_user = current_user
+    else
+      the_claims = claims
+      raise "Unauthenticated" unless the_claims    # Unauthenticated
+      if user = User.find_by(email: the_claims[0]['user']) and
```

Comment:
Style: Prefer `&&` unless there is a strong reason to use `and`. [[source](https://github.com/rubocop-hq/ruby-style-guide#user-content-no-and-or-or)]

Hmmmn. Also, I would alter `User.find_by(email:  foo)`, which returns an array (also, by the way but no important now, `#where` is preferred, stylistically), to `User.find_by_email(foo)`, which returns a single instance (the first), and thus you could avoid using `[0]` on line 35 below.

Reply:
re `&&`, ok, I don't understand this, but will accept without question. I was just copying other code I think.
re `find_by_email`, didn't know about that. will switch